common/web/gesture-recognizer/src/cumulativePathStats.ts

```diff
@@ -0,0 +1,702 @@
+        return Number.NaN;
+      }
+
+      const xDelta = this.lastSample.targetX - this.initialSample.targetX;
+      const yDelta = this.lastSample.targetY - this.initialSample.targetY;
+      const yAngleDiff = Math.acos(-yDelta / this.netDistance);
+
+      return xDelta < 0 ? (2 * Math.PI - yAngleDiff) : yAngleDiff;
+    }
+
+    /**
+     * Returns the angle (in degrees) traveled by the corresponding segment clockwise
+     * from the unit vector <0, -1> in the DOM (the unit "upward" direction).
+     */
+    public get angleInDegrees() {
+      return this.angle * 180 / Math.PI;
+    }
+
+    /**
+     * Returns the cardinal or intercardinal direction on the screen that most
+     * closely matches the direction of movement represented by the represented
+     * segment.
+     *
+     * @return A string one or two letters in length.  (e.g:  'n', 'sw')
+     */
+    public get cardinalDirection() {
+      if(this.sampleCount == 1 || !this.lastSample || !this.initialSample) {
+        return undefined;
+      }
+
+      const angle = this.angleInDegrees;
+      const buckets = ['n', 'ne', 'e', 'se', 's', 'sw', 'w', 'nw'];
+
+      for(let threshold = 22.5, bucketIndex = 0; threshold < 360; threshold += 45, bucketIndex += 1) {
+        if(angle < threshold) {
+          return buckets[bucketIndex];
+        }
+      }
+
+      return 'n';
+    }
+
+    /**
+     * Measured in pixels per second.
+     */
+    public get speed() {
+      // this.duration is already in seconds, not milliseconds.
+      return this.duration ? this.netDistance / this.duration : Number.NaN;
+    }
+
+    /**
+     * Returns the represented interval's 'mean angle' clockwise from the DOM's
+     * <0, -1> (the unit vector toward the top of the screen) in radians.
+     *
+     * Based upon the 'circular mean'.  Refer to https://en.wikipedia.org/wiki/Circular_mean.
+     *
+     * Note that very slow-moving segments may be heavily affected by pixel aliasing
+     * effects; mouse and touch events usually do not provide sub-pixel resolution.
+     */
+    public get angleMean() {
+      if(this.arcSampleCount == 0) {
+        return Number.NaN;
+      }
+
+      // Neato reference: https://rosettacode.org/wiki/Averages/Mean_angle
+      // But we don't actually need to divide by sample count; `atan2` handles that!
+      const sinMean = this.sinLinearSum;
+      const cosMean = this.cosLinearSum;
+
+      let angle = Math.atan2(sinMean, cosMean);  // result:  on the interval (-pi, pi]
+      // Convert to [0, 2*pi).
+      if(angle < 0) {
+        angle = angle + 2 * Math.PI;
+      }
+
+      return angle;
+    }
+
+    /**
+     * Provides the rSquared value needed internally for circular-statistic properties.
+     *
+     * Range:  floating-point values on the interval [0, 1].
+     */
+    private get angleRSquared() {
+      // https://www.ebi.ac.uk/thornton-srv/software/PROCHECK/nmr_manual/man_cv.html may be a useful
```

Comment:
Uh... that's a bit of an ask.  I'm currently unable to explain exactly why the definition is what it is, unlike with normal regression formulas and expressions - it's new to me and seems to have a more complex derivation than standard linear regression.  I can at least understand enough to interpret it and utilize it, though.

Upon another look, I can at least make https://en.wikipedia.org/wiki/Directional_statistics#Distribution_of_the_mean work here instead; might take a little clarification of how things map, but the parallel is about as clear.

Granted, yeah, it's still a link, but at least Wikipedia is good at taking steps to prevent link-rot.